lib/util/dump-collection.ts

```diff
@@ -0,0 +1,337 @@
+import ndjson from 'ndjson';
+import { JSONObject } from '../../index';
+import get from 'lodash/get';
+import isObject from 'lodash/isObject';
+import stream from 'stream';
+
+/**
+ * Flatten an object transform:
+ * {
+ *  title: "kuzzle",
+ *  info : {
+ *    tag: "news"
+ *  }
+ * }
+ *
+ * Into an object like:
+ * {
+ *  title: "kuzzle",
+ *  info.tag: news
+ * }
+ *
+ * @param {Object} target the object we have to flatten
+ * @returns {Object} the flattened object
+ */
+function flattenObject (target: JSONObject): JSONObject {
+  const output = {};
+
+  flattenStep(output, target);
+
+  return output;
+}
+
+function flattenStep (
+  output: JSONObject,
+  object: JSONObject,
+  prev: string | null = null): void {
+  const keys = Object.keys(object);
+
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    const value = object[key];
+    const newKey = prev ? prev + '.' + key : key;
+
+    if (Object.prototype.toString.call(value) === '[object Object]') {
+      output[newKey] = value;
+      flattenStep(output, value, newKey);
+    }
+
+    output[newKey] = value;
+  }
+}
+
+
+/**
+ * An iteration-order-safe version of lodash.values
+ *
+ * @param object The object containing the values
+ * @param fields The field names to pick in the right order
+ * @returns The values in the same order as the fields
+ * @see https://lodash.com/docs/4.17.15#values
+ */
+function pickValues (object: any, fields: string[]): any[] {
+  return fields.map(f => formatValueForCSV(get(object, f)));
+}
+
+/**
+ * Formats the value for correct CSV output, avoiding to return
+ * values that would badly serialize in CSV.
+ *
+ * @param value The value to format
+ * @returns The value or a string telling the value is not scalar
+ */
+function formatValueForCSV (value: any) {
+  if (isObject(value)) {
+    return '[NOT_SCALAR]';
```

Comment:
Maybe we could replace this by `[OBJECT] ` because this term is more common than scalar. Especially because users that will manipulate the CSV will not always be developers